src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +113,68 @@ public ArmSubsystem() {
         }
 
         // Set default PID values elbow
-        for (var config : new PID[]{ELBOW_DEFAULT, ELBOW_STEADY}) {
+        for (var config : new PID[] {ELBOW_DEFAULT, ELBOW_STEADY}) {
             m_elbow.getPIDController().setP(config.P, config.slot);
             m_elbow.getPIDController().setI(config.I, config.slot);
             m_elbow.getPIDController().setD(config.D, config.slot);
             m_elbow.getPIDController().setFF(config.FF, config.slot);
             m_elbow.getPIDController().setDFilter(config.DF, config.slot);
         }
 
+        m_elbow.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_elbow.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kReverse, true);
+
         // Set the soft limits
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
+    }
 
+    public void setState(State desiredState) {
+        this.m_desiredState = desiredState;
     }
 
-    public void setState(State state) {
-        // Don't do anything already in correct state
-        if (state == m_state_actual)
-            return;
-
-        switch (m_state_actual) {
-            case Pickup:
-                if (state == State.Rest)
-                    m_state_desired = state;
-                break;
-            case Rest:
-                m_state_desired = state;
-                break;
-            case PlacingA:
-            case PlacingB:
-            case PlacingC:
-            case PlacingD:
-            case PlacingE:
-                m_state_desired = state;
-                break;
-            case Transit:
-                break;
-        }
+    public boolean atState() {

```

Comment:
Where is this used in control?

Reply:
I hadn't gotten to it but it should be used in the arm setting command. It's very useful because it allows the command to finish only when the arm is at state, which is esp. useful for autos.